dvc/progress.py

```diff
@@ -33,43 +35,72 @@ class Tqdm(tqdm):
     maximum-compatibility tqdm-based progressbars
     """
 
+    BAR_FMT_DEFAULT = (
+        "{percentage:3.0f}%|{bar:10}|{desc} {bar:-10b}{n}/{total}"
+        " [{elapsed}<{remaining}, {rate_fmt:>11}{postfix}]"
+    )
+    BAR_FMT_NOTOTAL = (
+        "{desc} {bar:b}{n} [{elapsed}<??:??, {rate_fmt:>11}{postfix}]"
+    )
+
     def __init__(
         self,
         iterable=None,
         disable=None,
-        bytes=False,  # pylint: disable=W0622
+        level=logging.ERROR,
+        desc=None,
         desc_truncate=None,
         leave=None,
+        level_leave=logging.DEBUG,
+        bar_format=None,
+        bytes=False,  # pylint: disable=W0622
         **kwargs
     ):
         """
         bytes   : shortcut for
             `unit='B', unit_scale=True, unit_divisor=1024, miniters=1`
-        desc_truncate  : like `desc` but will truncate to 10 chars
+        desc  : persists after `close()`
+        desc_truncate  : like `desc` but will `truncate()` and not persist
+        level  : effective logging level for determining `disable`;
+            used only if `disable` is unspecified
+        level_leave  : effective logging level for determining `leave`;
+            used only if `leave` is unspecified
         kwargs  : anything accepted by `tqdm.tqdm()`
         """
         kwargs = deepcopy(kwargs)
+        kwargs.setdefault("unit_scale", True)
         if bytes:
             for k, v in dict(
                 unit="B", unit_scale=True, unit_divisor=1024, miniters=1
             ).items():
                 kwargs.setdefault(k, v)
+        if desc is not None:
+            self.desc_persist = desc
```

Comment:
Simply
```python
self.desc_persist = desc
```
Then `if self.desc_persist is not None: ...` in the end, instead of `hasattr()`.

Reply:
Also how does it works with `desc_truncate`? This asymmetry looks weird.

Reply:
I thought about it and it seems to make sense to support both even though it looks weird. `desc_truncate` would be for intermediate titles (which are also expected to be possibly long file names), whereas `desc` is the overall description which should also persist upon `close()`.

Reply:
removed now